semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/DocumentOps.scala

```diff
@@ -12,16 +12,17 @@ trait DocumentOps { self: DatabaseOps =>
     if (!g.settings.Yrangepos.value) {
       sys.error("the compiler instance must have -Yrangepos enabled")
     }
-    if (g.useOffsetPositions) {
-      sys.error("the compiler instance must use range positions")
-    }
+    // for some reason this fails with ammonite, even though the first Yrangepos == true
```

Comment:
I'm gonna need some help on this ... reading the code, `Yrangepos.value == true` and `useOffsetPositions` should be equivalent

Reply:
Oh right, I had missed the initialisation order. Guess I was spoiled by referential transparency 😆 

This does not look like something that's fixable on the scalameta side. If you're happy with the rest of the changes, I could uncomment these lines and move my efforts to the ammonite side. 

What do you think ?

Reply:
> Oh right, I had missed the initialisation order. Guess I was spoiled by referential transparency 😆

Welcome to `Global` :joy:


> This does not look like something that's fixable on the scalameta side. If you're happy with the rest of the changes, I could uncomment these lines and move my efforts to the ammonite side.

Indeed, this does not seem fixable on the scalameta side. The `Settings` that are passed to the `Global` constructor need to have Yrangepos enabled.